articles/howtos/create-new-decorator.md

```diff
@@ -0,0 +1,222 @@
+---
+uid: CreateNewDecorator
+---
+# How can I create a new Decorator?
+
+Click [here](xref:Decorator) to find out more about Decorators.
+
+This how-to will serve as an example of how to create a new Decorator that will inject code at runtime for a Template that has been setup for it.
+
+This how-to will build up from the [previous](xref:MakeTemplateDecoratorReady) how-to, which showed how to make a Template ready for Decorators. Please interpolate on your end what is written here to make the necessary adjustments on what you're trying to achieve.
+
+## Create a new Decorator
+
+Typically you would create a new Decorator in a new Intent Architect Application which will reference a Template from another Module or Intent Architect Application. As such we will be doing that.
+
+![Startup](images/create-new-decorator/MyDecoratorModuleStartup.png)
+
+Go ahead and create your Application in Intent Architect that will be used to create the Module that will contain the new Decorator for the `StartupTemplate`.
```

Comment:
```suggestion
Start by creating an Application in Intent Architect that will be used to create the Module that will contain the new Decorator for the `StartupTemplate`.
```

Reply:
That TIP doesn't explain how to make a module, only how to take an existing Template and make it Decorator ready. Might want to point it to a tutorial? Happy to change the code above.

Reply:
This relates to the comment above and my overall thinking is about aiming to overall keep this article as concise possible. The purpose of the tip is that "this is our starting point, check there for how to get there." ,

Reply:
```suggestion
Start by creating an Application in Intent Architect that will be used to create the Module that will contain the new Decorator for the `StartupTemplate`.

> [!HINT]
> If you want to know how to make a template decorator capable, refer to the [](xref:MakeTemplateDecoratorReady) how-to article.
```

By moving the link to the previous article to rather be a hint, it changes it from feeling like it's pre-requisite reading, to rather being optional reading. For readers who already know they need to implement a decorator, they can ignore this hint.

What do you think?